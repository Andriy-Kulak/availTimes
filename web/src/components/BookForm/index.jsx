import React from 'react'
import PropTypes from 'prop-types'
import Header from '../Header'

const BookForm = React.memo(({ onChange }) => {
  return (
    <>
      <Header isBig text="Book Time with an Advisor" />
      <form id="name-form" className="col-md-6">
        <div className="form-group">
          <label htmlFor="name-field">Your Name</label>
          <input
            type="text"
            id="name-field"
            className="form-control"
            onChange={(e) => onChange(e.target.value)}
          />
        </div>
      </form>
    </>
  )
})

BookForm.propTypes = {
  onChange: PropTypes.func,
}

export default BookForm
