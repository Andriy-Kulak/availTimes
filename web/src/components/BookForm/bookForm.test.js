import React from 'react'
import BookForm from './index'
import renderer from 'react-test-renderer'

describe('Book Form', () => {
  it('component matches', () => {
    const component = renderer.create(<BookForm onChange={() => ({})} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
