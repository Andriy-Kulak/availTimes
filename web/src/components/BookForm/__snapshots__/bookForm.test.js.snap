// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Form component matches 1`] = `
Array [
  <h1>
    Book Time with an Advisor
  </h1>,
  <form
    className="col-md-6"
    id="name-form"
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="name-field"
      >
        Your Name
      </label>
      <input
        className="form-control"
        id="name-field"
        onChange={[Function]}
        type="text"
      />
    </div>
  </form>,
]
`;
